components/forms/sql/create_shared_schema.sql

```diff
@@ -0,0 +1,101 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+CREATE TABLE IF NOT EXISTS addresses_data (
+    guid          TEXT NOT NULL PRIMARY KEY,
+    given_name          TEXT NOT NULL,
+    additional_name     TEXT NOT NULL,
+    family_name         TEXT NOT NULL,
+    organization        TEXT,                -- Company
+    street_address      TEXT,                -- (Multiline)
+    address_level3      TEXT NOT NULL,       -- Suburb/Sublocality
+    address_level2      TEXT NOT NULL,       -- City/Town
+    address_level1      TEXT NOT NULL,       -- Province (Standardized code if possible)
+    postal_code         TEXT NOT NULL,
+    country             TEXT NOT NULL,       -- ISO 3166
+    tel                 TEXT,                -- Stored in E.164 format
+    email               TEXT,
+
+    -- TODO: removed computed fields, need to figure out
+
+    time_created        INTEGER NOT NULL,
+    time_last_used      INTEGER,
+    time_last_modified  INTEGER NOT NULL,
+    times_used          INTEGER NOT NULL DEFAULT 0,
+
+    sync_change_counter INTEGER NOT NULL DEFAULT 1
+);
+
+CREATE TABLE IF NOT EXISTS addresses_mirror (
+    guid          TEXT NOT NULL PRIMARY KEY,
+    given_name          TEXT NOT NULL,
```

Comment:
I think these may need to be nullable, though, because we [filter out](https://searchfox.org/mozilla-central/rev/30e70f2fe80c97bfbfcd975e68538cefd7f58b2a/browser/extensions/formautofill/FormAutofillStorage.jsm#1269-1273) values with empty strings on Desktop, so they won’t show up in the incoming records at all.

Reply:
Or we could default them to empty strings, and/or get serde to do the right thing, right? (again, not really advocating one way or the other, but failing to see a semantic difference, and I see potential bugs where we assume `Some(s)` means `s` actually has a value when it may not.

Reply:
Sold! 😁